pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -928,20 +939,39 @@ func (r *ReconcilePerconaServerMongoDB) reconcileMongos(cr *api.PerconaServerMon
 		return nil
 	}
 
-	msDepl := psmdb.MongosDeployment(cr)
-	err = setControllerReference(cr, msDepl, r.scheme)
-	if err != nil {
-		return errors.Wrapf(err, "set owner ref for deployment %s", msDepl.Name)
-	}
-
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: msDepl.Name, Namespace: msDepl.Namespace}, msDepl)
-	if err != nil && !k8serrors.IsNotFound(err) {
-		return errors.Wrapf(err, "get deployment %s", msDepl.Name)
-	}
+	var mongos runtime.Object
+	if cr.CompareVersion("1.12.0") >= 0 {
+		msDepl := psmdb.MongosDeployment(cr)
+		err = r.client.Delete(context.TODO(), msDepl)
```

Comment:
as I understood this code is running on each reconcile loop, so
write operations are expensive on etcd, it is better to get the object and delete it only if it exists